test/TESTING_WITH_DOCKER.md

```diff
@@ -0,0 +1,131 @@
+# Testing With Docker
+
+These instructions will guide you through the process of setting up Docker
+to perform tests while developing on the New Relic Agent. Doing so can provide
+you with a consistent experience shared with other developers that helps to
+avoid developer computer specific issues.
+
+*NOTE:* Currently, fewer than 100% of the functional tests pass when ran under
+Docker. We are working to address the related compatibility issues.
+
```

Comment:
What do you think about creating a call to contribute at this line? We could put any outstanding issues within a milestone or give them a certain tag to generate a link to the information so that people reading this can help if interested?

Reply:
Terrific idea. I'm going to circle back to this one soon once I know the state of things. For multiverse we're down to a single suite not passing. For the different Rails dev environments, they haven't been tested yet.

Reply:
A new `all_compatible_envs` task has been added that will run all env tests for the current `RUBY_VERSION` based on the mappings defined in `.github/workflows/ci.yml`, and that task itself has been added to the list of what `test:all` does. With the entirety of all tests represented by that `test:all` task now passing with the default Ruby, I'm thinking we're ok without having to make a disclaimer or call for contributions.